src/AutoRest.CSharp/Common/Output/Models/Types/ObjectTypeProperty.cs

```diff
@@ -118,7 +118,7 @@ internal string CreateExtraDescriptionWithManagedServiceIdentity()
         {
             var extraDescription = string.Empty;
             var originalObjSchema = SchemaProperty?.Schema as ObjectSchema;
-            var identityTypeSchema = originalObjSchema?.GetAllProperties().FirstOrDefault(p => p.SerializedName == "type").Schema;
+            var identityTypeSchema = originalObjSchema?.GetAllProperties()?.FirstOrDefault(p => p.SerializedName == "type")?.Schema;
```

Comment:
Lets use `!` here instead of `?` so that if these are ever null the error is thrown from the same spot as before